test/cli/test-more-projects.py

```diff
@@ -224,4 +225,19 @@ def test_project_missing_subproject(tmpdir):
     ret, stdout, stderr = cppcheck(['--project=' + project_file, '--template=cppcheck1'])
     assert ret == 1, stdout
     assert stdout == "cppcheck: error: failed to open project '{}/dummy.json'. The file does not exist.\n".format(str(tmpdir).replace('\\', '/'))
-    assert stderr == ''
\ No newline at end of file
+    assert stderr == ''
+
+
+def test_hang_array_many_strings(tmpdir):
+    # 11901
+    # cppcheck valueflow hangs when analyzing a file with many strings
+    filename = os.path.join(tmpdir, 'hang.c')
+    with open(filename, 'wt') as f:
+        f.write("const char *strings[] = {\n")
+        for i in range(20000):
+            f.write('    "abc",\n')
+        f.write("};\n")
+    start_time = time.time()
```

Comment:
Please use `@pytest.mark.timeout()` - requires the `pytest-timeout` Python package to be added. This makes sure we don't hang indefinitely if this regresses again.

Also this is the wrong test file. This for for testing `--project=` inputs. Better use `test-others.py`.

This might be bound to fail (erroneously) if we have a particular slow runner in the CI. We should evaluate if we can use some kind of metric to check this in a better way (does not have to be fixed in this PR).

Reply:
alright I have added pytest-timeout.

> This might be bound to fail (erroneously) if we have a particular slow runner in the CI.

I hope the timeout 10 seconds is long enough. It takes ~1 second on my computer.

However another metric wouldn't hurt.